Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -1012,22 +1012,65 @@ template <typename TYPE> std::vector<DateAndTime> TimeSeriesProperty<TYPE>::time
  */
 template <typename TYPE>
 std::vector<DateAndTime> TimeSeriesProperty<TYPE>::filteredTimesAsVector(const Kernel::TimeROI *roi) const {
```

Comment:
I'm assuming the selection of the entries is the same as in `TimeSeriesProperty<TYPE>::filteredValuesAsVector` so would you please duplicate the docstring of that method in regards to which entries are returned?
```/**
 * Series values within the ROI's regions.
 * A values will be excluded if its times lie outside the ROI's regions or coincides with the upper boundary
 * of a ROI region. For instance, time "2007-11-30T16:17:30" is excluded in the ROI
 * [2007-11-30T16:17:00", 2007-11-30T16:17:30").
 * @param roi :: ROI regions validating any query time.
 * @returns :: Vector of included values only.
 */
template <typename TYPE> std::vector<TYPE> TimeSeriesProperty<TYPE>::filteredValuesAsVector(const TimeROI *roi) const {
```
and yes, I cringed a little when I saw the duplicate code. Would you consider a private method `std::vector<TimeValueUnit<TYPE>> filteredEntriesAsVector()` so that you can use it to retrieve either the values or the times?

Reply:
The bottom of the method is the main place things differ. If you want to cringe even more, look at `TimeSeriesProperty::findIndex()`.

I'll update the docs because they aren't quite right anymore. These methods generate the v6.6 fake values on the fly.

Reply:
so you think we should keep the duplicate code?

Reply:
@peterfpeterson , @jmborr , what do you think about the filtering approach used [here](https://github.com/mantidproject/mantid/blob/ewm1355_mean_math/Framework/Kernel/src/TimeSeriesProperty.cpp#L322-L336). Is it easy to follow?

Reply:
> @peterfpeterson , @jmborr , what do you think about the filtering approach used [here](https://github.com/mantidproject/mantid/blob/ewm1355_mean_math/Framework/Kernel/src/TimeSeriesProperty.cpp#L322-L336). Is it easy to follow?

Did you change this method? I vaguely recall the removal was done in-place, without having to create the temporary object `mpcopy` :thinking:

Reply:
No, it was like this from the get-go.  This is a refactoring of the old method - see the link: [TimeSeriesProperty<TYPE>::filterByTimes](https://github.com/mantidproject/mantid/blob/f3c86d4eeedd2763cfe7940083003e130abf5a85/Framework/Kernel/src/TimeSeriesProperty.cpp#L312-L383)